api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java

```diff
@@ -156,4 +164,8 @@ protected void handleHasAndListParam(Criteria criteria, HasAndListParam hasAndLi
 	
 	protected abstract Criterion generateNotCompletedOrderQuery(String path);
 	
+	protected abstract Criterion generateFulfillerStatusRestriction(String path, String fulfillerStatus);
```

Comment:
My only real suggestion here is that maybe these should be non-abstract and just do nothing? That way, they get implemented where we care about them (basically FhirEncounterDaoImpl_2_2), but can be skipped in other classes where the implementation don't make sense?

Reply:
Make sense, will update...